xapian-applications/omega/handler_libextractor.cc

```diff
@@ -74,12 +74,13 @@ process_metadata(void* cls,
 	case EXTRACTOR_METATYPE_ORIGINAL_TITLE:
 	case EXTRACTOR_METATYPE_SUBJECT:
 	case EXTRACTOR_METATYPE_TITLE:
-	    md->title += string(data, data_len);
-	    md->title += " ";
+	    if (!md->title.empty())
+		md->title.append(" ");
+	    md->title.append(data, data_len);
 	    break;
 
 	case EXTRACTOR_METATYPE_PAGE_COUNT:
-	    md->pages = string(data, data_len);
+	    md->pages.append(data, data_len);
```

Comment:
If we get multiple `EXTRACTOR_METATYPE_PAGE_COUNT` items (which seems like it would happen if more than one plugin reports a number of pages for a format) then this will now concatenate them as strings - e.g. if we're told 12 pages twice, we'll turn that into `1212`!

The behaviour before of taking the last answer in case of duplicates seems better, so I think you want the "assign" method instead of "append" here.

Reply:
Thanks for pointing this out. I have changed this.